internal/archive/archive_test.go

```diff
@@ -373,14 +459,20 @@ func (s *S) checkArchitecture(c *C, arch string, binaryPath string) {
 	c.Assert(binaryArch, Equals, arch)
 }
 
-func (s *S) testOpenArchiveArch(c *C, arch string) {
+func (s *S) testOpenArchiveArch(c *C, release ubuntuRelease, arch string) {
+	c.Logf("Checking ubuntu archive %s %s...", release.name, arch)
+
 	options := archive.Options{
 		Label:      "ubuntu",
-		Version:    "22.04",
+		Version:    release.version,
 		Arch:       arch,
-		Suites:     []string{"jammy"},
+		Suites:     []string{release.name},
 		Components: []string{"main", "universe"},
 		CacheDir:   c.MkDir(),
+		PublicKeys: []*packet.PublicKey{
+			ubuntuArchiveSignKey2018.PublicKey,
+			ubuntuArchiveSignKey2012.PublicKey,
+		},
```

Comment:
Can we include the public keys in each test? We should not be using the wrong key for each release.